tests/Functional/AutoincrementColumnTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+
+use function strtolower;
+
+class AutoincrementColumnTest extends FunctionalTestCase
+{
+    /** @var bool */
+    private static $tableCreated = false;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (! self::$tableCreated) {
+            $table = new Table('auto_increment_table');
+            $table->addColumn('id', 'integer', ['autoincrement' => true]);
+            $table->addColumn('test_int', 'integer');
+            $table->setPrimaryKey(['id']);
+
+            $sm = $this->connection->getSchemaManager();
+            $sm->createTable($table);
+            self::$tableCreated = true;
+        }
+
+        $platform = $this->connection->getDatabasePlatform();
+
+        $this->connection->executeStatement(
+            $platform->getTruncateTableSQL('auto_increment_table')
+        );
+    }
+
+    public function testInsertArbitraryId(): void
+    {
+        if ($this->connection->getDatabasePlatform()->getName() === 'mssql') {
+            self::markTestSkipped('Test does not work on MSSql.');
+        }
+
+        $this->connection->insert('auto_increment_table', ['test_int' => 1]);
+
+        $this->assertEquals([
+            ['id' => 1, 'test_int' => 1],
+        ], $this->lowerCaseResultColumnNames($this->connection->fetchAllAssociative(
```

Comment:
`lowerCaseResultColumnNames()` isn't necessary. If a single column is selected, you can use `fetchOne()`, and we don't need to select all columns.

Reply:
Changed to use `fetchFirstColumn()`.